community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/ParallelSort.java

```diff
@@ -345,25 +343,36 @@ private void qsort( int initialStart, int initialEnd )
             }
         }
 
-        private int informedPivot( int start, int end, int randomIndex )
+        private long randomLong( long diff )
+        {
+            // It pays to have this extra check ...
+            return diff > Integer.MAX_VALUE
+                    // ... since this call is more than twice as expensive as ...
+                    ? Math.abs( random.nextLong() ) % diff
+                    // ... this one
+                    : random.nextInt( (int) diff );
+        }
```

Comment:
Odd. It has more instructions. They should at best have the same performance. Could be random variance or a figment of the JIT compiler.